### PR TITLE
Extract all the layers included inner layers when doing "load_weights…

### PR DESCRIPTION
**Pull request by edwardyehuang**
For example, a model included:
      -Conv2D (keras.layer)
      -BatchNorm (Keras.layer)
      -CustomLayer (keras.layer)
      -CustomModel (keras.Model)
          -Conv2D (keras.layer)
          -Conv2D (keras.layer)
          -etc..
      -Conv2D (keras.layer)

This function will extract all the layers included the layers in CustomModel or deeper. This is helpful when load a weights by name from functional model h5 file to subclass model.

### CODE REVIEW

**Comment by edwardyehuang** [comment]
#29310

**Comment by edwardyehuang** [comment]
I signed it!

File: tensorflow/python/keras/engine/network.py [hunk 1]
```diff
@@ -1357,10 +1357,40 @@ def load_weights(self, filepath, by_name=False):
       if 'layer_names' not in f.attrs and 'model_weights' in f:
         f = f['model_weights']
       if by_name:
-        saving.load_weights_from_hdf5_group_by_name(f, self.layers)
+        saving.load_weights_from_hdf5_group_by_name(f, self._get_all_layers(self))
       else:
         saving.load_weights_from_hdf5_group(f, self.layers)
 
+  def _get_all_layers (self, network):
```
Comment (tanzhenyu) [inline tensorflow/python/keras/engine/network.py]: Can you add tests for sequential & functional & subclass models?

File: tensorflow/python/keras/engine/network.py [hunk 2]
```diff
@@ -1357,10 +1357,40 @@ def load_weights(self, filepath, by_name=False):
       if 'layer_names' not in f.attrs and 'model_weights' in f:
         f = f['model_weights']
       if by_name:
-        saving.load_weights_from_hdf5_group_by_name(f, self.layers)
```
Comment (tanzhenyu) [inline tensorflow/python/keras/engine/network.py]: Don't you need to do this for save_weights as well?

Comment (edwardyehuang) [reply tensorflow/python/keras/engine/network.py]: Yea. But in this I commit I did not consider about the save weights. Because usually we want to move the name based weights to object based. I will add it.

File: tensorflow/python/keras/engine/network.py (hunk 1 above)
Comment (edwardyehuang) [reply tensorflow/python/keras/engine/network.py]: Will do ti

**Comment by gbaned** [comment]
@edwardyehuang Any update on reviewer comments? Please.

**Comment by edwardyehuang** [comment]
> 
> 
> @edwardyehuang Any update on reviewer comments? Please.

Sorry, I am very busy recently. I will finish it by 6 sept

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by edwardyehuang** [comment]
Hi, sorry. I am still working on it. Hopefully finish in this week. Very busy on August.

**Comment by gbaned** [comment]
@edwardyehuang Could you please resolve the conflicts? Thanks!

**Comment by edwardyehuang** [comment]
> 
> 
> @edwardyehuang Could you please resolve the conflicts? Thanks!

Resolved

**Comment by gbaned** [comment]
@edwardyehuang Could you please resolve the conflicts? Thanks!

**Comment by edwardyehuang** [comment]
> @edwardyehuang Could you please resolve the conflicts? Thanks!

Resolved

**Review by tanzhenyu — DISMISSED** [review]

**Comment by gbaned** [comment]
@edwardyehuang Could you please address Ubuntu Sanity errors? Thanks!

**Comment by edwardyehuang** [comment]
tensorflow/python/keras/engine/network.py:1202: [C0301(line-too-long), ] Line too long (116/80)

tensorflow/python/keras/engine/network.py:1206: [C0326(bad-whitespace), ] No space allowed before bracket

Should be fixed

**Comment by gbaned** [comment]
@tanzhenyu Can you please take a look on this PR? Thanks!

**Review by tanzhenyu — COMMENTED** [review]
Do you have an existing example on how the current code would fail, but the new code would succeed, i.e., unit test?
On the other hand, this seems backward in-compatible though, it would break many code IIUC.

**Comment by gbaned** [comment]
@edwardyehuang Can you please check tanzhenyu's comments and keep us posted. Thanks!
